chore: creation of the history page (FE-1)

edgarolg opened the pull request:
Creation of the main history page. This is so the team knows where it is, and so they can play here and do all the things here.

It is a basic page, everyone will be adding their things.

Main modifications:

1. The Endpoint card can now be used to display the hour.
2. The Button component has been changed so it can handle de goto function.
3. There's a new dynamic page for the user's full history, depending on the time range the user selects.
4. Leave some TODOS for personal reminders. This will be done on a different PR, so the team can have the changes and can start working with something more accurate.


For the reviewers: the main changes are on the following files
1. Endpoint card
2. History (everything on the folder)
3. Button

The other files are because of the rebase to fix the issue with the automatic tests, hope you see this :)

LouArcTec commented:
LGTM

edgarolg committed:
chore: creation of the history page (FE-1)

edgarolg committed:
chore: creation of the history page (FE-1)

edgarolg committed:
chore: fix design for card and layout (FE-1)

edgarolg committed:
chore: fix design for card and layout (FE-1)

edgarolg commented:
@DevEquinox o @LouArc / @LouArcTec  oyes, me sale error en un test, pero está raro pq sale ese, puesto que es para probar la página principal y necesita que funcione el api wraper, pero ya intente de que mockearla y asi, sigue sin funcionar, help!!

Se puede borrar la pr, puesto que son cosas peques so no pasa nada si se quita o así

DevEquinox commented:
Muy buenas
Ahí lo reviso

edgarolg commented:
Va, como quiera si algo dime que es y ahí veo moverle o borro el test jeje

DevEquinox commented:
Smn
La intención es decirles los errores, pero que los commits sean suyos, para que no anden apareciendo coauthors innecesarios

DevEquinox commented:
Oh
Ya entendí qué estás intentando mockear
@LouArc ApiWrapper ocupa que el backend esté disponible de forma local (usa localhost:8080), pero está fallando la conexión en el test en el repo porque obviamente no viene incluido el setup de frontend

Deja busco alternativas

edgarolg commented:
sisi, pero lo que se me hace raro es que yo no hice ese test ni nada, ya estaba, pero nomas a mí me fallo

y si intente hacer de que un mock, pero aun así fallaba, como que de fuerza lo quiere de verdad, hice algo mas o menos así

`vi.stubGlobal('fetch', vi.fn().mockResolvedValue({
	ok: true,
	json: async () => [
		{ id: 1, name: 'Mocked Service' },
		{ id: 2, name: 'Another Service' }
	],
}));`

DevEquinox commented:
Ah, ya sé por qué pasó al repo a pesar de estar fallando (si no tienes prendido el back).
En el husky se corre únicamente los tests de playwright.
No consideré vitest, y según yo lo había excluido del GitHub Actions.
Al momento de subir las cosas, como aún no estaban listas las branch rules, no se corrió el npm test (y según yo había dejado únicamente `npx playwright test`).

edgarolg commented:
Ohhh, vaa

Entonces que hacemos, hacemos para corregir el test o lo agregamos al ignore de los test el vite o nos hacemos locos y no hacemos el reto??

DevEquinox commented:
Creo que voy a tener que abrir una branch y ticket para cada equipo para arreglar ese pedo nosotros
De ahí te recomiendo hacer rebase de tu branch. Por lo que estoy viendo, no debería de haber conflictos, y como es tu branch personal (no compartida) no debería de causar problemas como los que causó en el repo viejito.

edgarolg commented:
Vaa, muchas graciaaas!

DevEquinox committed:
chore: see description (SCRUM-65) (#4)

Implement mock API, and migrate the API into App Container useContext in pages to not recreate an ApiWrapper.
Add extension of typescript's Window to allow storing an important variable Remove incorrect page test (anything UI related is through playwright. Speaking of, see future comment in related issue).
Configure test projects so we can now run a mock version and a live (for integration tests) version of our project.
Added separate folders for mock and integration tests.
Change npm test and the husky test to only run the mock version (since backend is unavailable).
Add app.d.ts to tsconfig so the contents (like interface) are recognized.
Kill Icon component due to performance and redundancy

DevEquinox committed:
chore: see description (SCRUM-65) (#6)

Implement mock API, and migrate the API into App Container useContext in pages to not recreate an ApiWrapper. Add extension of typescript's Window to allow storing an important variable Remove incorrect page test (anything UI related is through playwright. Speaking of, see future comment in related issue). Configure test projects so we can now run a mock version and a live (for integration tests) version of our project. Added separate folders for mock and integration tests. Change npm test and the husky test to only run the mock version (since backend is unavailable). Add app.d.ts to tsconfig so the contents (like interface) are recognized. Kill Icon component due to performance and redundancy

DevEquinox committed:
chore: bring fixes to team-3-stable (SCRUM-65) (#9)

DevEquinox commented:
Buenas
¿Podrías probar un `git pull origin/team-2-stable --rebase` o un `git rebase team-3-stable`, y luego un `git push --force` para cambiar tus commits para que se basen en la nueva versión y reescriban la nueva versión? De esa forma deberían de volver a correr los checks de cada commit que falló.

edgarolg committed:
chore: Merge branch 'FE-1' of github.com:CustomerAutoZoneSelfService/cazss-frontend into (FE-1)

edgarolg committed:
chore: little change to see if fix (FE-1)

DevEquinox reviewed (dismissed):
👍

edgarolg committed:
chore: bones of the pages for history (FE-1)

DevEquinox commented on src/lib/components/Button.svelte:
Hi
Since Svelte 5:
- Props are recommended over export let (unless you're exporting types and functions to import in other files, but in the case of replacing variables in a component, `$props()` is more convenient)
- If you use props, you can skip createEventDispatcher altogether.

You might prefer using props to import the `onClick` function, and just rely on the if `!disabled`.
And if I remember correctly (at least when I tested it using props), if `onClick` is empty, it just wont do anything, so you don't have to worry about it throwing an error in that case.

You can see the following for the alternatives:
https://svelte.dev/docs/svelte/v5-migration-guide

DevEquinox commented on src/routes/history/+page.svelte:
This page fails due to the following:
```
Svelte error: each_key_duplicate
Keyed each block has duplicate key `999` at indexes 0 and 1
```

The reason is clear: the 999 is being reused as a key (not allowed), but it might not be the kind of key you're thinking of. Keys in this context might be referring to a value that helps Svelte identify which individual components it should rerender if something changes in some cases. It isn't recommended to just use an array's id (or an equivalent) unless you know it's something absolutely unique.
In this case I'd recommend you do what Ernesto did in his `InputTable` component's `forEach`s. He uses `index` after the value he acceses in his for each, which is a sort of implicit variable that contains the current element's `index`, but specifying `(index)` doesn't actually result in Svelte trying to use that. Instead, it tells Svelte to stick to what it does by default.

Visit https://svelte.dev/docs/svelte/each for the info in the official documentation.

DevEquinox replied:
Though Svelte understands this when you don't specify anything in the parenthesis too, TypeScript might leave some warnings asking you to state it explicitly (which I think is what happened to Ernesto).

DevEquinox replied:
I'd be lying if I told you I knew _what Svelte's default key mechanism is_ when you specify (index) (or nothing), but the point is it tells Svelte to take care of it and it takes care of the TypeScript warning.

edgarolg replied:
Yes, I did know that the keys must be unique, but this is handle like this because they are just for display, when the real endpoints of the DB are used, it will have a unique Id, this will be the history_Id or the exact name used in the DB.

For practical purpose I let this like this, so the team can work better, but I do intend to make this correctly (leave a to-do comment, so I don't forget)

DevEquinox reviewed (changes requested):
See the comments.

edgarolg committed:
chore: requested changes (FE-1)

DevEquinox reviewed (dismissed):
Looks good

LouArcTec reviewed (commented):
@DevEquinox is his husky modification allowed?

LouArcTec commented:
Please resolve merge conflicts in branch

edgarolg commented:
@LouArcTec The husky is because of the rebase of the branch, but don't worry, because if you merge the other one, all my relevant changes are already in the stable, so the merge of this branch may be not necessary

edgarolg commented:
This was closed because the changes were merged on another branch, to know more info ask me.
